### PR TITLE
Non-empty combinators return NonEmptyList

### PR DESCRIPTION
**Pull request by robertdp**
For https://github.com/purescript-contrib/purescript-parsing/issues/83

### CODE REVIEW

File: src/Text/Parsing/Parser/Combinators.purs [hunk 1]
```diff
@@ -99,30 +100,33 @@ lookAhead p = (ParserT <<< ExceptT <<< StateT) \s -> do
...
 sepBy1 p sep = do
   a <- p
   as <- many $ sep *> p
-  pure (a : as)
+  pure (cons' a as)
 
 -- | Parse phrases delimited and optionally terminated by a separator.
 sepEndBy :: forall m s a sep. Monad m => ParserT s m a -> ParserT s m sep -> ParserT s m (List a)
-sepEndBy p sep = sepEndBy1 p sep <|> pure Nil
+sepEndBy p sep = (toList <$> sepEndBy1 p sep) <|> pure Nil
```
Comment (robertdp) [inline src/Text/Parsing/Parser/Combinators.purs:114]: Would it be beneficial to separate these related combinators, so the non-empty combinators don't depend on their possibly-empty counterparts and vice-versa?

Comment (thomashoneyman) [reply src/Text/Parsing/Parser/Combinators.purs:114]: Yes, I think that’s sensible.

Comment (robertdp) [reply src/Text/Parsing/Parser/Combinators.purs]: Updated.

File: src/Text/Parsing/Parser/Combinators.purs [hunk 2]
```diff
@@ -193,8 +204,9 @@ manyTill p end = scan
...
 
 -- | Parse several phrases until the specified terminator matches, requiring at least one match.
-many1Till :: forall s a m e. Monad m => ParserT s m a -> ParserT s m e -> ParserT s m (List a)
+many1Till :: forall s a m e. Monad m => ParserT s m a -> ParserT s m e -> ParserT s m (NonEmptyList a)
 many1Till p end = do
   x <- p
   xs <- manyTill p end
-  pure (x:xs)
+  pure (cons' x xs)
+
```
Comment (thomashoneyman) [inline src/Text/Parsing/Parser/Combinators.purs:212]:
```suggestion
```

**Review by thomashoneyman — APPROVED** [review]
Thank you!

**Comment by jamesdbrock** [comment]
Thank you @robertdp !

**Pull request merged**